lib/smart_proxy_reports/spooled_http_client.rb

```diff
@@ -59,11 +59,16 @@ def process
       # send all files via a single persistent HTTP connection
       logger.debug "Opening HTTP connection to Foreman"
       client.http.start do |http|
-        Dir.glob(spool_path("todo", "*")) do |filename|
+        Dir.glob([spool_path("todo", "facts*"), spool_path("todo", "reports*")]) do |filename|
```

Comment:
Looks like you divide this with `-` (a dash), I suggest to include it: `"facts-*"` (or singular).

One thing is that by providing an array to glob, you are essentially concatenating both (different) things you want to do into a single array:

```ruby
>> Dir.glob(["a*", "b*"])
=> ["app", "babel.config.js"]
```

You basically "throw away" this intel and then down below you need to have an if-statement to get the intel back. If you would extract most of the code from the loop into helper methods, then you can simply do this:

```ruby
Dir.glob("facts...") do |f|
  call_helper_method(:fact, filename)
end
Dir.glob("reports...") do |f|
  call_helper_method(:report, filename)
end
```

In the helper method you can simply decide via an argument.